docs/polkadot/networking/3-avail-valid.md

```diff
@@ -230,57 +233,42 @@ When receiving each piece, we also need to be able to authenticate it individual
+It's unnecessary to distinguish between "is a validator" and "acting as a proxy for a validator". This could be given as optional information in the certificate (e.g. so that the peer expects a higher latency), or it may be omitted if the validator wants to withhold this information from its peers. Nodes are free to guess whether their peers are proxies or not.
 
-If a proxy is chosen successfully, this should be remembered for the next period and the proxy may be re-used, skipping any failed attempts to send directly or via other proxies. Of course, if this proxy subsequently fails then we can remove this association and retry the steps from scratch.
+#### Proxy protocol
 
-The above applies for all stages where distributors are active. As per the [summary table](#Protocol-overview) this is all stages, except phase 2 stage B.
+The proxying protocol is straightforward, since the private validator node and the sentry nodes trust each other.
 
-**Distributees**
+1. Inbound, the protocol does not require any special headers (unwrapping/rewrapping of the content). Whenever a sentry node accepts an incoming connection, it forwards it directly onto the single sentry node.
 
-As per the [summary table](#Protocol-overview) distributees are only active in stage B.
+    Justification: in our A&V direct-sending protocol, the contents are all signed by their authors, so there is no need for extra checking at the sentry node, although this may be done either to simplify the code or as extra "defense in depth". In all cases, proper exercise of flow control at the private validator node is necessary to prevent the sentry node from spamming it by mistake.
 
-For phase 1 stage B, we have N/C possible options to ask for every single piece.
+2. Outbound, the protocol needs special headers for the private validator node to tell the sentry node the outgoing destination. This is straightforward: namely the peer validator's key.
 
-For phase 2 stage B, we have N possible options to ask for ceil(N/3) pieces.
+Recall that as above, there are two types of outbound connections: distributors pushing data, or distributees requesting data.
 
-These numbers are high enough that we consider it unnecessary to specify that requests-for-pieces can be proxied, which itself is also slightly more complex than proxying a push as it involves one extra half-round of messages. This allows us to avoid some (hopefully) unnecessary complexity in what is already a fairly complex protocol.
+Since the private validator node may not be able to access the address book, the sentry node is the one to perform the address book lookup. As described in 1(a) above, in the general case it will get a set of addresses as the result. For better load-balancing, the sentry node should sort this set and select the jth address to connect to, where j = i mod n, n is the size of the set, and (c, i) is the co-ordinate of its validator.
```

Comment:
```suggestion
Say the address book is realized via a Kademlia Dht. Given that the private validator won't be able to make outbound connections beyond its sentry nodes the validator node may not be able to access the address book. The sentry node is the one to perform the address book lookup. As described in 1(a) above, in the general case it will get a set of addresses as the result. For better load-balancing, the sentry node should sort this set and select the jth address to connect to, where j = i mod n, n is the size of the set, and (c, i) is the co-ordinate of its validator.
```

Reply:
I wanted this paragraph to be general, since in the future we may want to have part of the address-book onchain in which case the private validator node would be able to access it. But still part of it would be on the kademlia DHT, so it is probably better to have the sentry node resolve it all the time.